flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/ReplaceIntersectWithSemiJoinRule.scala

```diff
@@ -49,7 +49,8 @@ class ReplaceIntersectWithSemiJoinRule
 
     val relBuilder = call.builder
     val keys = 0 until left.getRowType.getFieldCount
-    val conditions = generateEqualsCondition(relBuilder, left, right, keys)
+    val conditions =
+      generateEqualsCondition(relBuilder, left, right, keys.map(Int.box).toList.asJava)
```

Comment:
NIT: use `Integer.valueOf` instead?

Reply:
@JingGe 
we can see two style in many scala Rules files like
    keys.map(Int.box).toList.asJava, in CommonPhysicalLookupJoin/AggregateUtil
   map(Integer.valueOf).toList.asJava in many files. 
so i change it to your suggestion